docs/en/upgrade.md

```diff
@@ -51,7 +55,70 @@ You can view the status of the stack in the AWS CloudFormation console in the **
+
+### Migrate the Existing Data (only applicable when upgrading from a version earlier than v1.1.6)
+
+!!! info "Important"
+
+    The data migration process is CPU-intensive. Before starting the migration, ensure that the load on your Redshift is low. It's also advisable to consider temporarily increasing the RPUs of Redshift Serverless or the cluster size when migrating large volumes of data.
+
+    In our benchmark, we migrated 100 million events in 25 minutes using 32 RPUs of Redshift Serverless.
+
+1. Open [Redshift query editor v2][query-editor]. You can refer to the AWS document [Working with query editor v2][working-with-query-editor] to log in and query data using Redshift query editor v2.
+
+    !!! info "Note"
+        You must use the `admin` user or a user with schema (known as the app ID) ownership permission. See this [FAQ][view-schema-in-redshift] for more details.
+
+2. Select the Serverless workgroup or provisioned cluster, `<project-id>`->`<app-id>`->Tables, and ensure that tables for the appId are listed there.
+
+3. Create a new SQL Editor, select your project's schema.
+
+4. Customize the date range as desired, and execute the following SQL in the editor to migrate events from the past 180 days, or any number of days up to the present, to the new tables.
+
+    ```sql
+    -- please replace `<app-id>` with your actual app id
+    -- update the day range based on your needs
+    CALL "<app-id>".sp_migrate_data_to_v2(180);
+    ```
+
+5. Wait for the SQL to complete. The execution time depends on the volume of data in the `events` table.
+
+6. Execute the following SQL to check the stored procedure execution log; ensure there are no errors. If there are any interruptions, timeouts, or other errors, you can re-execute step 4 to continue the data migration.
+
+    ```sql
+    -- please replace `<app-id>` with your actual app id
+    SELECT * FROM "<app-id>"."clickstream_log" WHERE log_name = 'sp_migrate_event_to_v2' ORDER BY log_date DESC;
+    SELECT * FROM "<app-id>"."clickstream_log" WHERE log_name = 'sp_migrate_user_to_v2' ORDER BY log_date DESC;
+    SELECT * FROM "<app-id>"."clickstream_log" WHERE log_name = 'sp_migrate_item_to_v2' ORDER BY log_date DESC;
+    SELECT * FROM "<app-id>"."clickstream_log" WHERE log_name = 'sp_migrate_session_to_v2' ORDER BY log_date DESC;
+    SELECT * FROM "<app-id>"."clickstream_log" WHERE log_name = 'sp_migrate_data_to_v2' ORDER BY log_date DESC;
+    ```
+
+7. Calculate the metrics for the migrated data used in the out-of-the-box dashboards. Refer to [this FAQ][faq-recalculate-data] for guidance.
```

Comment:
--> 7. Follow this guide [this FAQ][faq-recalculate-data] to calculate metrics for the new preset dashboard based on the migrated data.